File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minProfit = float('inf')
        maxProfit = 0
        for i in range(len(prices)):
            if prices[i] < minProfit:
                minProfit = prices[i]
            elif prices[i] - minProfit > maxProfit:
                maxProfit = prices[i] - minProfit
        return maxProfit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minProfit = INT_MAX;
        int maxProfit = 0;
        for(auto price: prices){
            if(price < minProfit){
                minProfit = price;
            }else if(price-minProfit > maxProfit){
                maxProfit = price-minProfit;
            }
        }
        return maxProfit;
    }
};